packages/cli/src/commands/replay/utils/compute-and-save-diff.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  ScreenshotAssertionsEnabledOptions,
+  ScreenshotDiffResult,
+} from "@alwaysmeticulous/api";
+import { METICULOUS_LOGGER_NAME } from "@alwaysmeticulous/common";
+import { AxiosInstance } from "axios";
+import log from "loglevel";
+import { createReplayDiff } from "../../../api/replay-diff.api";
+import {
+  getOrFetchReplay,
+  getOrFetchReplayArchive,
+  getReplayDir,
+  getScreenshotsDir,
+} from "../../../local-data/replays";
+import {
+  checkScreenshotDiffResult,
+  diffScreenshots,
+  ScreenshotDiffError,
+} from "../../screenshot-diff/screenshot-diff.command";
+
+export interface ComputeAndSaveDiffOptions {
+  client: AxiosInstance;
+  testRunId: string | null;
+  baseReplayId: string;
+  headReplayId: string;
+  tempDir: string;
+  screenshottingOptions: ScreenshotAssertionsEnabledOptions;
+}
+
+export const computeAndSaveDiff = async ({
+  client,
+  baseReplayId,
+  tempDir,
+  headReplayId,
+  screenshottingOptions,
+  testRunId,
+}: ComputeAndSaveDiffOptions): Promise<{
+  screenshotDiffResults: ScreenshotDiffResult[];
+  screenshotDiffError: ScreenshotDiffError | null;
+}> => {
+  const logger = log.getLogger(METICULOUS_LOGGER_NAME);
```

Comment:
This is mostly just cut and pasted out from replay.command.ts (replay.command.ts was getting pretty big)